src/publishProject/publishProjectWebViewController.ts

```diff
@@ -319,6 +354,67 @@ export class PublishProjectWebViewController extends FormWebviewController<
         );
     }
 
+    /** Auto-select a new connection and populate server/database fields */
+    private async autoSelectNewConnection(connectionUri: string): Promise<void> {
+        try {
+            // IMPORTANT: Get the connection profile FIRST, before any async calls
+            // The connection URI may be removed from activeConnections during async operations
+            const connection = this._connectionManager.activeConnections[connectionUri];
+            if (!connection || !connection.credentials) {
+                return; // Connection not found or no credentials available
+            }
+
+            const connectionProfile = connection.credentials as IConnectionProfile;
+            if (!connectionProfile || !connectionProfile.server) {
+                return; // Connection profile invalid or no server specified
+            }
+
+            // Update server name immediately
+            this.state.formState.serverName = connectionProfile.server;
+
+            // Get connection string first
+            const connectionString = await this._connectionManager.getConnectionString(
+                connectionUri,
+                true, // includePassword
+                true, // includeApplicationName
+            );
+            this.state.connectionString = connectionString;
+
+            // Get databases
+            const databases = await this._connectionManager.listDatabases(connectionUri);
```

Comment:
Maybe a simple follow up task. If the list Database API fails, maybe show an error message to the user.